diff --git a/drivers/rtc/rtc-ds1302.c b/drivers/rtc/rtc-ds1302.c
index 2f83adef9..1a83965bc 100644
--- a/drivers/rtc/rtc-ds1302.c
+++ b/drivers/rtc/rtc-ds1302.c
@@ -7,6 +7,7 @@
  */
 
 #include <linux/bcd.h>
+#include <linux/delay.h>
 #include <linux/init.h>
 #include <linux/io.h>
 #include <linux/kernel.h>
@@ -109,6 +110,9 @@ static int ds1302_probe(struct spi_device *spi)
 	u8		*bp;
 	int		status;
 
+	/* Allow CE to remain inactive before the first transaction. */
+	usleep_range(10000, 12000);
+
 	/* Sanity check board setup data.  This may be hooked up
 	 * in 3wire mode, but we don't care.  Note that unless
 	 * there's an inverter in place, this needs SPI_CS_HIGH!
