--- drivers/ata/pata_ixp4xx_cf.c.orig	2025-07-17 16:30:55.000000000 +0000
+++ drivers/ata/pata_ixp4xx_cf.c	2026-08-17 01:17:09.527938597 +0000
@@ -261,6 +261,19 @@
 	dev_info(dev, "using CS%d for PIO timing configuration\n", csindex);
 	ixpp->cmd_csreg = csindex * IXP4XX_EXP_TIMING_STRIDE;
 
+	/* Determine whether the board provides a CF interrupt. */
+	irq = platform_get_irq_optional(pdev, 0);
+
+	if (irq > 0) {
+		irq_set_irq_type(irq, IRQ_TYPE_EDGE_RISING);
+	} else if (irq == -ENXIO || irq == -ENODEV) {
+		dev_info(dev, "no IRQ, using PIO polling\n");
+		irq = 0;
+		pi.flags |= ATA_FLAG_PIO_POLLING;
+	} else if (irq < 0) {
+		return irq;
+	}
+
 	ixpp->host = ata_host_alloc_pinfo(dev, ppi, 1);
 	if (!ixpp->host)
 		return -ENOMEM;
@@ -275,20 +288,12 @@
 	if (IS_ERR(ixpp->cmd) || IS_ERR(ixpp->ctl))
 		return -ENOMEM;
 
-	irq = platform_get_irq(pdev, 0);
-	if (irq > 0)
-		irq_set_irq_type(irq, IRQ_TYPE_EDGE_RISING);
-	else if (irq < 0)
-		return irq;
-	else
-		return -EINVAL;
-
 	/* Just one port to set up */
 	ixp4xx_setup_port(ixpp->host->ports[0], ixpp, cmd->start, ctl->start);
 
 	ata_print_version_once(dev, DRV_VERSION);
 
-	return ata_host_activate(ixpp->host, irq, ata_sff_interrupt, 0, &ixp4xx_sht);
+	return ata_host_activate(ixpp->host, irq, irq ? ata_sff_interrupt : NULL, 0, &ixp4xx_sht);
 }
 
 static const struct of_device_id ixp4xx_pata_of_match[] = {
